### ISSUE TITLE
Add "Force push with lease" to the push dialog

### ISSUE THREAD

**Issue by otbutz** [gitextensions/gitextensions#3934]
**Do you want to request a *feature* or report a *bug*?**
Feature

**What is the current behavior?**
The Push dialog only offers the option to "Force push" which can be quite [dangerous](https://developer.atlassian.com/blog/2015/04/force-with-lease/)

![2017-08-24 11_23_19-push was rejected from _origin_](https://user-images.githubusercontent.com/22315436/29659791-961a3a70-88bf-11e7-86fd-3c4d40b63d79.png)

**What is the expected behavior?**
Add the option "Force push with lease" to provide more safety for users who often push rebased branches.

**Comment by EbenZhang** [comment]
it does force with lease if your git supports it. I think we Can reword it to force with lease and drop the support of old version of git, probably prior to 1.9 cannot remember for sure

**Comment by otbutz** [comment]
Git supports it since 1.8.5 ([Release notes](https://github.com/git/git/blob/35f6318d44379452d8d33e880d8df0267b4a0cd0/Documentation/RelNotes/1.8.5.txt#L182-L184))

Dropping support for git versions prior to 1.9 seems reasonable IMHO.

**Comment by RussKie** [comment]
Is anyone willing to put a PR for this?
